opcua/client/client.py

```diff
@@ -288,7 +303,10 @@ def send_hello(self):
         Send OPC-UA hello to server
         """
         ack = self.uaclient.send_hello(self.server_url.geturl(), self.max_messagesize, self.max_chunkcount)
-        # FIXME check ack
+
+        # TODO: Handle ua.UaError
+        if isinstance(ack, ua.UaStatusCodeError):
+            raise ack
```

Comment:
this might be correct, I am not sure, I do not remember what send_hello returns...

Reply:
A successful `send_hello` returns an [Acknowledge Message](https://github.com/FreeOpcUa/python-opcua/blob/master/opcua/ua/uaprotocol_hand.py#L81-L97). Looking at Figure 13 and 14 in OPC UA Part 6, 7.1.3, when the client receives the acknowledge message, it'll send a `Open Secure Channel Request`. Not sure if this is helpful, but in the logs, I see this as the header for the `ACK` message:

```
2018-12-05 13:50:31.292 [opcua.uaprotocol] 123145336487936 INFO - received header: Header(type:b'ACK', chunk_type:b'F', body_size:20, channel:0)
```

On a side note, is it fair to assume that a [`UaError`](https://github.com/FreeOpcUa/python-opcua/blob/master/opcua/ua/uaerrors/_base.py#L23-L24) is a client-side error?

Reply:
UaError in an exception created from an OPC UA Error, you can raise that Error on client or server